### DISCUSSION TITLE
Chat id

### DISCUSSION THREAD

**Discussion by semernur** [def-studio/telegraph#290] [Ideas]
Wouldn't it be better if we pass chat_id every time we use send method while sending a message? is there a way to do so?

**Comment by fabio-ivona** [comment]
Chat id is usually taken from the TelegraphChat model, or the methods are directly called from the model

Could you elaborate your needs?

**Comment by semernur** [comment]
![tg](https://user-images.githubusercontent.com/85545098/207199658-14c5a192-56ee-4e20-89e6-3bfc2c919524.png)

One of the packages [ https://github.com/irazasyed/telegram-bot-sdk/ ] I have been using uses the above style which is so simple I guess. We don't need to execute a query to find to which chat we are sending the message. We get the chat_id from every update so we don't need the query at all.

**Reply by fabio-ivona** [reply]
That package uses a low-level API to Telegram, this one was built around the concept of Models, but I think we can implement a way to force a chat id without using the DB, thanks for the suggestion

**Reply by semernur** [reply]
That is why I am trying to shift to your package. The sooner implementation will be better I am waiting your next release.
By the way I loved the documentation

**Reply by fabio-ivona** [reply]
Thanks! docs took a lot of effort but I'm pretty happy with the result, though I'd like to make some sections more clear

**Reply by fabio-ivona** [reply]
@semer-11 implemented, would you mind to test it? 

#294

**Reply by semernur** [reply]
Wow!

Is it included in the docs

**Reply by fabio-ivona** [reply]
not yet as it is not released, but in the PR you find all you need

**Comment by semernur** [comment]
Okay send me how I can do that please

**Reply by fabio-ivona** [reply]
two ways:

you add these keys in config/telegraph.php

```
[

    'bot_token' => 'AAAABBBB',
    'chat_id' => '-123456789',

]
```

or you hint bot and chat for each call:

```
Telegraph::bot('AAAAABBBB')->chat('-123456789')->message('hi')->send();
```

**Comment by semernur** [comment]
Awesome
Thank you

**Comment by semernur** [comment]
Hi
It throws exception when I try to use

Telegraph::bot('token')->chat(123456787)->message("Hello")->send();

The exception
TypeError: DefStudio\Telegraph\Telegraph::bot(): Argument #1 ($bot) must be
of type DefStudio\Telegraph\Models\TelegraphBot, string given, called in
C:\Users\user\Desktop\Project\sofonias\Sofonias\vendor\laravel\framework\src\Illuminate\Support\Facades\Facade.php
on line 338 in file
C:\Users\user\Desktop\Project\sofonias\Sofonias\vendor\defstudio\telegraph\src\Concerns\HasBotsAndChats.php
on line 33

**Reply by fabio-ivona** [reply]
Hi, will check in a few hours

**Reply by fabio-ivona** [reply]
@semer-11 which version of telegraph did you installed?

**Comment by semernur** [comment]
I have just installed it as it shows in the doc

**Comment by semernur** [comment]
I guess it's is v1.26.2

**Reply by fabio-ivona** [reply]
for under-development feature a dev- branch is required, that's why you did not see the new feature

try requiring "dev-feat-bypass-telegraph-models" version

**Reply by fabio-ivona** [reply]
just replace it to the telegraph version number inside your composer.json and run composer update
